File: java/src/main/java/it/gabrieletondi/telldontaskkata/domain/Category.java
package it.gabrieletondi.telldontaskkata.domain;

import java.math.BigDecimal;

public class Category {
    private String name;
    private final BigDecimal taxPercentage;

    public Category(String name, BigDecimal taxPercentage) {
        this.name = name;
        this.taxPercentage = taxPercentage;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getTaxPercentage() {
        return taxPercentage;
    }

}

File: typescript/src/domain/Category.ts
class Category {
  //private name: string;
  private readonly taxPercentage: number;

  constructor(name: string, taxPercentage: number) {
    //this.name = name;
    this.taxPercentage = taxPercentage;
  }

  public getTaxPercentage(): number {
    return this.taxPercentage;
  }

}

export default Category;

